controllers/cluster_controller.go

```diff
@@ -435,9 +435,44 @@ func (r *FoundationDBClusterReconciler) updateOrApply(ctx context.Context, clust
 	return r.Status().Update(ctx, cluster)
 }
 
+// getAllPodStatus will get the current status and phase of all the pods in the fdb cluster and check if all of them are up before proceeding with
+// connecting to fdb cluster and get information on coordinator servers, etc. Once all pods are up and running it will return true else false.
+func (r *FoundationDBClusterReconciler) getAllPodStatus(ctx context.Context, logger logr.Logger, cluster *fdbv1beta2.FoundationDBCluster) bool {
+	processCounts, err := cluster.GetProcessCountsWithDefaults()
+	logger.Info("Cluster Info", "Cluster", cluster, "Process Counts", processCounts)
+	if err != nil {
+		return false
+	}
+
+	for _, processGroup := range cluster.Status.ProcessGroups {
+		if processGroup.IsMarkedForRemoval() {
+			logger.V(1).Info("Ignore process group marked for removal",
+				"processGroupID", processGroup.ProcessGroupID)
+			continue
+		}
+
+		pod, err := r.PodLifecycleManager.GetPod(ctx, r, cluster, processGroup.GetPodName(cluster))
+		// If a Pod is not found ignore it for now.
```

Comment:
The comment is actually wrong, if a Pod is not found the whole method will return false. We probably can remove this comment.